sklearn/decomposition/_fastica.py

```diff
@@ -585,7 +608,13 @@ def g(x, fun_args):
 
         self.n_iter_ = n_iter
 
-        if self.whiten:
+        if self._whiten:
+            if self._whiten == "unit-variance":
+                S = np.linalg.multi_dot([W, K, XT]).T
```

Comment:
If `compute_sources` is True, this is computed twice.

Reply:
4a4c3c3 horribly solves the problem. Should the logic here be simplified?

Reply:
I don't have a better suggestion from the top of my head.